packages/hash/frontend/src/components/block-loader/block-loader.tsx

```diff
@@ -1,44 +1,41 @@
-import { UnknownRecord } from "@blockprotocol/core";
 import {
-  BlockGraph,
   BlockGraphProperties,
-  EntityType as BpEntityType,
+  EntityEditionId,
+  GraphResolveDepths,
+  Subgraph,
+  SubgraphRootTypes,
 } from "@blockprotocol/graph";
+import { VersionedUri } from "@blockprotocol/type-system/slim";
 import { HashBlockMeta } from "@hashintel/hash-shared/blocks";
-import { JsonSchema } from "@hashintel/hash-shared/json-utils";
-import { EntityId } from "@hashintel/hash-subgraph";
-import { uniqBy } from "lodash";
+import { EntityId } from "@hashintel/hash-shared/types";
+import { Subgraph as LocalSubgraph } from "@hashintel/hash-subgraph";
+import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
 import {
   FunctionComponent,
   useCallback,
+  useEffect,
   useLayoutEffect,
   useMemo,
   useRef,
+  useState,
 } from "react";
-import { useLocalstorageState } from "rooks";
 
 import { useBlockLoadedContext } from "../../blocks/on-block-loaded";
-import { useBlockContext } from "../../blocks/page/block-context";
-import { convertApiEntityToBpEntity } from "../../lib/entities";
 import { useIsReadonlyMode } from "../../shared/readonly-mode";
 import { useBlockProtocolAggregateEntities } from "../hooks/block-protocol-functions/knowledge/use-block-protocol-aggregate-entities";
+import { useBlockProtocolGetEntity } from "../hooks/block-protocol-functions/knowledge/use-block-protocol-get-entity";
+import { useBlockProtocolUpdateEntity } from "../hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
 import { useBlockProtocolFileUpload } from "../hooks/block-protocol-functions/use-block-protocol-file-upload";
 import { RemoteBlock } from "../remote-block/remote-block";
-import { DataMapEditor } from "./data-map-editor";
 import { fetchEmbedCode } from "./fetch-embed-code";
-import { SchemaMap } from "./shared";
 
-// @todo consolidate these properties, e.g. take all entityX, linkX into a single childEntity prop
-// @see https://app.asana.com/0/1200211978612931/1202807842439190/f
 type BlockLoaderProps = {
-  blockEntityId: string;
+  blockEntityId?: EntityId; // @todo make this always defined
+  blockEntityTypeId: VersionedUri;
   blockMetadata: HashBlockMeta;
-  blockSchema: JsonSchema;
   editableRef: (node: HTMLElement | null) => void;
-  entityId: EntityId;
-  entityTypeId: string;
-  entityProperties: {};
   onBlockLoaded: () => void;
+  wrappingEntityId: string;
```

Comment:
Is this not an actual `EntityId`?

Reply:
It should be, although as part of the wider 'block might be new and entities not created' problem it's possibly an empty string... to type it properly I'd have to either (a) assert it when loading this component, or (b) implement 'don't load this until entities are back from db', which I've no idea how much work it is

I can do (a) if you think it preferable 

Here's the relevant field being defined in the calling file
<img width="460" alt="Screenshot 2023-01-16 at 13 45 29" src="https://user-images.githubusercontent.com/37743469/212692648-dae863c8-85b5-46ef-a324-50e4e413fa7e.png">

Reply:
I don't have a good sense of what's worth the investment at the moment. Perhaps it's fine to leave as is. Just quite uncomfortable to see an `entityId` variable which is not an `EntityId`